### ISSUE TITLE
Error calling Leadership listener - Error on certificate generation on v1.5.4 / Consul KV

### ISSUE THREAD

**Issue by carlosedp** [containous/traefik#3345]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

I've deployed a Traefik cluster with Consul backend for clustering and certificate sharing between the nodes. The consul node was deployed with helm and is working perfectly.

After startup, I'm seeing this error:
```
time="2018-05-17T17:20:13Z" level=error msg="Error calling Leadership listener: get directory at 'https://acme-v01.api.letsencrypt.org/directory': failed to get json "https://acme-v01.api.letsencrypt.org/directory": Get https://acme-v01.api.letsencrypt.org/directory: x509: certificate is valid for 4054f16e279fdd3596300953c1a982a3.fe42775eeaedc065fcdb360e28027cd7.traefik.default, not acme-v01.api.letsencrypt.org"
```

I tried changing to API v2 but the same happens with Traefik 1.6.1.

This is the full startup log:

```
time="2018-05-17T17:06:36Z" level=info msg="Using TOML configuration file /config/traefik.toml"
time="2018-05-17T17:06:36Z" level=info msg="Traefik version v1.5.4 built on 2018-03-15_01:35:21PM"
time="2018-05-17T17:06:36Z" level=info msg="
Stats collection is disabled.
Help us improve Traefik by turning this feature on :)
More details on: https://docs.traefik.io/basics/#collected-data
"
time="2018-05-17T17:06:36Z" level=info msg="Preparing server http &{Network: Address::80 TLS:<nil> Redirect:0x4420293710 Auth:<nil> WhitelistSourceRange:[] Compress:true ProxyProtocol:<nil> ForwardedHeaders:0x442024afa0} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-05-17T17:06:36Z" level=info msg="Preparing server https &{Network: Address::443 TLS:0x44202ef680 Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] Compress:true ProxyProtocol:<nil> ForwardedHeaders:0x442024afc0} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-05-17T17:06:36Z" level=info msg="Starting server on :80"
time="2018-05-17T17:07:36Z" level=info msg="Starting server on :443"
time="2018-05-17T17:07:36Z" level=info msg="Preparing server traefik &{Network: Address::8080 TLS:<nil> Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] Compress:false ProxyProtocol:<nil> ForwardedHeaders:0x442024afe0} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-05-17T17:07:36Z" level=info msg="Starting provider *consul.Provider {"Watch":true,"Filename":"","Constraints":[],"Trace":false,"DebugLogGeneratedTemplate":false,"Endpoint":"consul-traefik-consul.kube-system.svc:8500","Prefix":"traefik-external","TLS":null,"Username":"","Password":""}"
time="2018-05-17T17:07:36Z" level=info msg="Starting provider *kubernetes.Provider {"Watch":true,"Filename":"","Constraints":[],"Trace":false,"DebugLogGeneratedTemplate":false,"Endpoint":"","Token":"","CertAuthFilePath":"","DisablePassHostHeaders":false,"EnablePassTLSCert":false,"Namespaces":null,"LabelSelector":"traffic-type=external"}"
time="2018-05-17T17:07:36Z" level=info msg="Creating in-cluster Provider client"
time="2018-05-17T17:07:36Z" level=info msg="Starting server on :8080"
time="2018-05-17T17:07:36Z" level=info msg="Node eaf9cb4d-2d1d-45cb-beb9-113519d08a1c elected worker ♝"
time="2018-05-17T17:07:37Z" level=error msg="Cannot unmarshall private key []"
time="2018-05-17T17:07:37Z" level=error msg="Error building ACME client &{Email: Registration:<nil> PrivateKey:[] DomainsCertificate:{Certs:[] lock:{w:{state:0 sema:0} writerSem:0 readerSem:0 readerCount:0 readerWait:0}} ChallengeCerts:map[] HTTPChallenge:map[]}: private key was nil"
time="2018-05-17T17:07:37Z" level=info msg="Node eaf9cb4d-2d1d-45cb-beb9-113519d08a1c elected leader ♚"
time="2018-05-17T17:07:37Z" level=info msg="Starting ACME renew job..."
time="2018-05-17T17:07:37Z" level=info msg="Server configuration reloaded on :80"
time="2018-05-17T17:07:37Z" level=info msg="Server configuration reloaded on :443"
time="2018-05-17T17:07:37Z" level=info msg="Server configuration reloaded on :8080"
time="2018-05-17T17:07:37Z" level=info msg="Skipping same configuration for provider consul"
time="2018-05-17T17:07:37Z" level=info msg="Skipping same configuration for provider consul"
time="2018-05-17T17:07:37Z" level=error msg="Datastore sync error: Object lock value: expected 73ee7515-8bd0-4452-9119-173b602475fb, got , retrying in 565.51565ms"
time="2018-05-17T17:07:37Z" level=info msg="Skipping same configuration for provider consul"
time="2018-05-17T17:07:37Z" level=error msg="Failed to retrieve basic auth configuration for ingress kube-system/kubernetes-dashboard-external: failed to load auth credentials: secret "kube-system"/"dashboard-auth" not found"
time="2018-05-17T17:07:37Z" level=error msg="Failed to retrieve basic auth configuration for ingress logging/kibana-external: failed to load auth credentials: secret "logging"/"ingress-auth" not found"
time="2018-05-17T17:07:37Z" level=error msg="Failed to retrieve basic auth configuration for ingress logging/cerebro-external: failed to load auth credentials: secret "logging"/"ingress-auth" not found"
time="2018-05-17T17:07:37Z" level=info msg="Server configuration reloaded on :443"
time="2018-05-17T17:07:37Z" level=info msg="Server configuration reloaded on :8080"
time="2018-05-17T17:07:37Z" level=info msg="Server configuration reloaded on :80"
time="2018-05-17T17:07:39Z" level=info msg="Server configuration reloaded on :443"
time="2018-05-17T17:07:39Z" level=info msg="Server configuration reloaded on :8080"
time="2018-05-17T17:07:39Z" level=info msg="Server configuration reloaded on :80"
time="2018-05-17T17:16:38Z" level=warning msg="Error checking new version: Get https://update.traefik.io/repos/containous/traefik/releases: x509: certificate is valid for 4054f16e279fdd3596300953c1a982a3.fe42775eeaedc065fcdb360e28027cd7.traefik.default, not update.traefik.io"
time="2018-05-17T17:20:13Z" level=error msg="Error calling Leadership listener: get directory at 'https://acme-v01.api.letsencrypt.org/directory': failed to get json "https://acme-v01.api.letsencrypt.org/directory": Get https://acme-v01.api.letsencrypt.org/directory: x509: certificate is valid for 4054f16e279fdd3596300953c1a982a3.fe42775eeaedc065fcdb360e28027cd7.traefik.default, not acme-v01.api.letsencrypt.org"

```

### What did you expect to see?

Success in clustering and certificate generation.

### What did you see instead?

Certificate is not generated. This error:

```
time="2018-05-17T17:20:13Z" level=error msg="Error calling Leadership listener: get directory at 'https://acme-v01.api.letsencrypt.org/directory': failed to get json "https://acme-v01.api.letsencrypt.org/directory": Get https://acme-v01.api.letsencrypt.org/directory: x509: certificate is valid for 4054f16e279fdd3596300953c1a982a3.fe42775eeaedc065fcdb360e28027cd7.traefik.default, not acme-v01.api.letsencrypt.org"
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

Traefik 1.5.4, Consul 1.1.0.

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

My configuration on a Configmap:

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: traefik-conf-external
  namespace: kube-system
data:
  traefik.toml: |
    defaultEntryPoints = ["http","https"]
    debug = false
    logLevel = "INFO"

    #Config to redirect http to https
    [entryPoints]
      [entryPoints.http]
        address = ":80"
        compress = true
        [entryPoints.http.redirect]
        entryPoint = "https"
      [entryPoints.https]
        address = ":443"
        compress = true
        [entryPoints.https.tls]

    [api]
      [api.statistics]
        recentErrors = 10

    [kubernetes]
      # Only create ingresses where the object has traffic-type: external label
      labelselector = "traffic-type=external"

    [metrics]
      [metrics.prometheus]
      buckets=[0.1,0.3,1.2,5.0]
      entryPoint = "traefik"

    [ping]
      entryPoint = "http"

    [accessLog]

    [consul]
      endpoint = "consul-traefik-consul.kube-system.svc:8500"
      watch = true
      prefix = "traefik-external"

    [acme]
      email = "myemail@email.com"
      storage = "traefik-external/acme/account"
      acmeLogging = true
      entryPoint = "https"
      OnHostRule = true
      caServer = "https://acme-v01.api.letsencrypt.org/directory"
      [acme.httpChallenge]
        entryPoint="http"
```
This gets loaded into KV by a initContainer on startup with "storeconfig" command.

This is the StatefulSet:

```yaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: traefik-ingress-controller-external
  namespace: kube-system
  labels:
    k8s-app: traefik-ingress-lb-external
spec:
  serviceName: traefik-ingress-lb-external
  replicas: 2
  selector:
    matchLabels:
      k8s-app: traefik-ingress-lb-external
  template:
    metadata:
      labels:
        k8s-app: traefik-ingress-lb-external
        name: traefik-ingress-lb-external
    spec:
      serviceAccountName: traefik-ingress-controller
      terminationGracePeriodSeconds: 60
      # Remove the initContainer in case of a *big operation* to avoid inconsistencies when
      # starting multiple pods in parallel. Might be a case a pod starts with partial config
      # because another pod is overwriting the config in KV.
      initContainers:
      - name: storeconfig
        image: traefik:v1.5.4
        imagePullPolicy: IfNotPresent
        args:
        - storeconfig
        - -c
        - /config/traefik.toml
        volumeMounts:
        - name: config
          mountPath: /etc/traefik
          readOnly: true
      containers:
      - image: traefik:v1.5.4
        name: traefik-ingress-lb
        ports:
         - name: http
           containerPort: 80
         - name: https
           containerPort: 443
         - name: admin
           containerPort: 8080
        args:
        - --api
        - --configfile=/config/traefik.toml
        livenessProbe:
          httpGet:
            path: /health
            port: 8080
          initialDelaySeconds: 180
          periodSeconds: 3
          timeoutSeconds: 1
        volumeMounts:
        - mountPath: /config
          name: config
      volumes:
      - name: config
        configMap:
          name: traefik-conf-external
      affinity:
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - weight: 100
            podAffinityTerm:
              labelSelector:
                matchExpressions:
                - key: role
                  operator: In
                  values:
                  - data
              topologyKey: kubernetes.io/hostname

```

**Comment by dtomcej** [comment]
If you look at the log you provided, there are a couple of interesting lines:

```
time="2018-05-17T17:16:38Z" level=warning msg="Error checking new version: Get https://update.traefik.io/repos/containous/traefik/releases: x509: certificate is valid for 4054f16e279fdd3596300953c1a982a3.fe42775eeaedc065fcdb360e28027cd7.traefik.default, not update.traefik.io"
time="2018-05-17T17:20:13Z" level=error msg="Error calling Leadership listener: get directory at 'https://acme-v01.api.letsencrypt.org/directory': failed to get json "https://acme-v01.api.letsencrypt.org/directory": Get https://acme-v01.api.letsencrypt.org/directory: x509: certificate is valid for 4054f16e279fdd3596300953c1a982a3.fe42775eeaedc065fcdb360e28027cd7.traefik.default, not acme-v01.api.letsencrypt.org"
```

It appears that the same traefik self signed certificate is being served for both `https://update.traefik.io` and `https://acme-v01.api.letsencrypt.org`.

I would hazard a guess that the issue here is your kube-dns, and routing ;) If both of those DNS lookups resolve to the same server ;)

**Comment by dtomcej** [comment]
@carlosedp Can you please confirm that you can curl those from the alpine container? It would ensure that DNS is not the issue.

**Comment by carlosedp** [comment]
Thanks for pointing this out @dtomcej, I found that the Alpine container messes up the name resolution. Look:

Pod: busybox created with:

```
apiVersion: v1
kind: Pod
metadata:
  name: busybox
  namespace: default
spec:
  containers:
  - name: busybox
    image: busybox
    command:
      - sleep
      - "3600"
    imagePullPolicy: IfNotPresent
  restartPolicy: Always

```

Resolution ok:
```
/ # cat /etc/resolv.conf
nameserver 10.96.0.10
search default.svc.cluster.local svc.cluster.local cluster.local internal.carlosedp.com
options ndots:5
/ #
/ # ping update.traefik.io
PING update.traefik.io (104.27.185.74): 56 data bytes
64 bytes from 104.27.185.74: seq=0 ttl=58 time=15.943 ms
64 bytes from 104.27.185.74: seq=1 ttl=58 time=13.562 ms
^C
--- update.traefik.io ping statistics ---
2 packets transmitted, 2 packets received, 0% packet loss
round-trip min/avg/max = 13.562/14.752/15.943 ms

/ # ping acme-v01.api.letsencrypt.org
PING acme-v01.api.letsencrypt.org (104.86.70.170): 56 data bytes
64 bytes from 104.86.70.170: seq=0 ttl=57 time=49.084 ms
64 bytes from 104.86.70.170: seq=1 ttl=57 time=60.945 ms
```


Pod: alpine created with:
```
apiVersion: v1
kind: Pod
metadata:
  name: alpine
  namespace: default
spec:
  containers:
  - image: alpine
    command:
      - /bin/sh
      - "-c"
      - "sleep 60m"
    imagePullPolicy: IfNotPresent
    name: alpine
  restartPolicy: Always

```

Resolution not OK:
```

/ # cat /etc/os-release
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.7.0
PRETTY_NAME="Alpine Linux v3.7"
HOME_URL="http://alpinelinux.org"
BUG_REPORT_URL="http://bugs.alpinelinux.org"
/ # cat /etc/resolv.conf
nameserver 10.96.0.10
search default.svc.cluster.local svc.cluster.local cluster.local internal.carlosedp.com
options ndots:5
/ #
/ # ping update.traefik.io
PING update.traefik.io (192.168.1.20): 56 data bytes
^C
--- update.traefik.io ping statistics ---
2 packets transmitted, 0 packets received, 100% packet loss
/ #
/ # ping acme-v01.api.letsencrypt.org
PING acme-v01.api.letsencrypt.org (192.168.1.20): 56 data bytes

^C
--- acme-v01.api.letsencrypt.org ping statistics ---
2 packets transmitted, 0 packets received, 100% packet loss
/ # ping google.com
PING google.com (192.168.1.20): 56 data bytes

^C
--- google.com ping statistics ---
2 packets transmitted, 0 packets received, 100% packet loss
```

I also have a CentOS container as a jumpbox into my system and it resolves perfectly as well.

Weird that both resolv.conf are identical but on Alpine, it resolves to my local DNS. This 192.168.1.20 IP is a wildcard DNS entry for *.internal.carlosedp.com. It's a dnsmasq service running on my router. It doesn't accept names without a domain:

```
rock64@kubemaster1:~ (kubearm:kube-system) $ ping blabla
ping: blabla: No address associated with hostname
rock64@kubemaster1:~ (kubearm:kube-system) $ ping blabla.internal.carlosedp.com
PING blabla.internal.carlosedp.com (192.168.1.20) 56(84) bytes of data.
```

Kubedns was deployed with no changes on the config. I also dropped into my kubedns container and the resolution is ok:

Pod: kube-dns-6448b967fc-2784w

```
Defaulting container name to kubedns.
Use 'kubectl describe pod/kube-dns-6448b967fc-2784w -n kube-system' to see all of the containers in this pod.
/ #
/ #
/ # cat /etc/resolv.conf
nameserver 192.168.1.1
search internal.carlosedp.com
/ #
/ # ping update.traefik.io
PING update.traefik.io (104.27.184.74): 56 data bytes
64 bytes from 104.27.184.74: seq=0 ttl=58 time=12.640 ms
^C
--- update.traefik.io ping statistics ---
2 packets transmitted, 1 packets received, 50% packet loss
round-trip min/avg/max = 12.640/12.640/12.640 ms
/ #
/ # ping acme-v01.api.letsencrypt.org
PING acme-v01.api.letsencrypt.org (104.86.70.170): 56 data bytes
64 bytes from 104.86.70.170: seq=0 ttl=57 time=50.298 ms
64 bytes from 104.86.70.170: seq=1 ttl=57 time=51.683 ms
^C
--- acme-v01.api.letsencrypt.org ping statistics ---
2 packets transmitted, 2 packets received, 0% packet loss
round-trip min/avg/max = 50.298/50.990/51.683 ms
```

**Comment by carlosedp** [comment]
Found out the problem was related to Alpine pods being scheduled into a new node I added to the cluster where it had the search path domains in it's resolv.conf. The pods were inheriting the config and couldn't resolve.

After I changed the host's resolv.conf to remove this search path, the containers started resolving correctly. Still it's a problem related to Alpine like mentioned in last comment.
